public/js/modMap.js

```diff
@@ -0,0 +1,343 @@
+            x: i * 20,
+            y: n * 60
+          }).data('key', {
+            x,
+            y
+          }).stroke({
+            color: '#5ECCA2',
+            width: 1
+          })
+        }
+        if (t["shape"] == "T3") {
+          if (t["orientation"] == "Ascend") {
+            var rect = draw.polygon('0,0 60,20 60,0').attr({
+              fill: defaultColor,
+              opacity: op,
+              x: i * 20,
+              y: n * 60
+            }).data('key', {
+              x,
+              y
+            }).stroke({
+              color: '#5ECCA2',
+              width: 1
+            })
+
+          }
+          if (t["orientation"] == "Decend") {
+            var rect = draw.polygon('0,0 60,20 60,0').attr({
+              fill: defaultColor,
+              opacity: op,
+              x: i * 20,
+              y: n * 60
+            }).data('key', {
+              x,
+              y
+            }).stroke({
+              color: '#5ECCA2',
+              width: 1
+            })
+
+          }
+          var rect = draw.rect(20, 60).attr({
+            fill: defaultColor,
+            opacity: op,
+            x: i * 20,
+            y: n * 60
+          }).data('key', {
+            x,
+            y
+          }).stroke({
+            color: '#5ECCA2',
+            width: 1
+          })
+
+        }
+        if (t["shape"] == "R3") {
+          var rect = draw.rect(20, 60).attr({
+            fill: defaultColor,
+            opacity: op,
+            x: i * 20,
+            y: n * 60
+          }).data('key', {
+            x,
+            y
+          }).stroke({
+            color: '#5ECCA2',
+            width: 1
+          })
+        }
+      } else {
+        defaultColor = 'white'
+        op = 1
+        var rect = draw.rect(20, 60).attr({
+          fill: defaultColor,
+          opacity: op,
+          x: i * 20,
+          y: n * 60
+        }).data('key', {
+          x,
+          y
+        }).stroke({
+          color: '#5ECCA2',
+          width: 1
+        })
+
+      }
+
+      var g = i * 20
+      var h = n * 60
+
+      // text.move(x: g, y: h).font({ fill: '#f06', family: 'Inconsolata' });
+
+
+
+      rect.click(function () {
+        document.getElementById("x").value = this.data('key')["x"];
+        document.getElementById("y").value = this.data('key')["y"];
+        modpage = 'module/'+this.data('key')["x"]+'&'+this.data('key')["y"];
+        window.location.href = modpage;
+        drawMod(t["shape"], t["id"])
+        this.fill({
+          color: color
+        })
+        var a = this.data('key')["n"]
+        console.log(a);
+        // plantMatrix.push({color: color, location:a });
+        //plantMatrix.push([color, a]);
+
+
+
+      })
+    }
+    var local = x
+    console.log(local);
+    var text = draw.text(local.toString()).attr({
+      opacity: op,
+      x: i * 20 - 6,
+      y: n * 60 + 12
+    }).rotate(90);
+  }
+
+};
+// function modMap() {
```

Comment:
do we want to keep this commented out?

Reply:
I actually should just delete it. The previous version used the plants to derive the modules placed which did not really make sense if we had empty modules

Reply:
yeah, I agree. if you need this code again, you have it in the commit history. let's avoid checking in commented code if possible